src/groupByKey.ts

```diff
@@ -2,6 +2,24 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce(
+    (acum: {}, item: T) => {
+      const NameKey = String(item[key]);
```

Comment:
we make a first letter capital only if it's a type. We don't use them for types
```suggestion
      const nameKey = String(item[key]);
```